File: prototype/rails/src/test/domain/reports/listing_status_count_test.rb
require "test_helper"

module Domain
  module Reports
    class ListingStatusCountTest < ActiveSupport::TestCase
      test "it labels its status for a dashboard tile" do
        assert_equal "For sale", ListingStatusCount.new(status: "for_sale", count: 3).label
      end
    end
  end
end

File: prototype/php/src/app/Domain/Reports/ListingStatusCountTest.php
<?php

declare(strict_types=1);

namespace App\Domain\Reports;

use App\Domain\Listings\ListingStatus;

it('carries a status and its count, and labels itself by the status', function (): void {
    $count = ListingStatusCount::of(ListingStatus::ForSale, 3);

    expect($count->status)->toBe(ListingStatus::ForSale)
        ->and($count->count)->toBe(3)
        ->and($count->label())->toBe('For sale');
});
